### PR TITLE
Feature/6030 logs

### PR DESCRIPTION
**Pull request by jesuspg**
#### REVIEWERS

@pratid
#### DESCRIPTION
- Add transaction Id
- Add debug traces
- Add support por docker to PhoneHome process
#### TESTING
- Locally and with docker

### CODE REVIEW

File: fiware-region-sanity-tests/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+RUN source /opt/fihealth/bin/activate; cd pygobject-2.20.0;./configure --prefix=/opt/fihealth
+RUN source /opt/fihealth/bin/activate; cd pygobject-2.20.0;make && make install
+
+RUN mkdir /opt/fiware-health
+ADD . /opt/fiware-health
+WORKDIR /opt/fiware-health
+RUN cp resources/docker/system.conf /etc/dbus-1/system.conf
+RUN pip2.7 install -r requirements.txt --allow-all-external
+EXPOSE 8081
+CMD ./resources/docker/start.sh
```
Comment (flopezag) [inline fiware-region-sanity-tests/Dockerfile]: add new line

File: fiware-region-sanity-tests/resources/docker/start.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+rm -rf /var/run/dbus/system_bus_socket;
+service messagebus restart
+export PYTHONPATH=$PWD
+export TEST_PHONEHOME_ENDPOINT=http://localhost:8081
+nohup python2.7 commons/http_phonehome_server.py > /var/log/httpPhoneHomeServer.log 2>&1&
+tail -f /var/log/httpPhoneHomeServer.log
```
Comment (flopezag) [inline fiware-region-sanity-tests/resources/docker/start.sh]: new line?

File: fiware-region-sanity-tests/resources/docker/system.conf [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+       holes in the above policy for specific services. -->
+  <includedir>system.d</includedir>
+
+  <!-- This is included last so local configuration can override what's
+       in this standard file -->
+  <include ignore_missing="yes">system-local.conf</include>
+
+  <include if_selinux_enabled="yes" selinux_root_relative="yes">contexts/dbus_contexts</include>
+
+</busconfig>
```
Comment (flopezag) [inline fiware-region-sanity-tests/resources/docker/system.conf]: new line

File: fiware-region-sanity-tests/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+RUN tar -xzf Python-2.7.6.tgz
+RUN cd Python-2.7.6;./configure --prefix=/usr/local --enable-unicode=ucs4 --enable-shared LDFLAGS="-Wl,-rpath /usr/local/lib"
+RUN cd Python-2.7.6;make && make altinstall
+RUN wget --no-check-certificate http://pypi.python.org/packages/source/d/distribute/distribute-0.6.35.tar.gz
+RUN tar xf distribute-0.6.35.tar.gz
+RUN cd distribute-0.6.35;python2.7 setup.py install
+RUN easy_install-2.7 pip
+#RUN rpm -ivh http://dl.fedoraproject.org/pub/epel/6/x86_64/epel-release-6-8.noarch.rpm
+RUN pip2.7 install virtualenv
+WORKDIR /opt
```
Comment (flopezag) [inline fiware-region-sanity-tests/Dockerfile]: Is it needed to fix again the WORKDIR?

File: fiware-region-sanity-tests/resources/docker/start.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+rm -rf /var/run/dbus/system_bus_socket;
```
Comment (pratid) [inline fiware-region-sanity-tests/resources/docker/start.sh]: Please remove semicolon

Comment (jesuspg) [inline fiware-region-sanity-tests/resources/docker/start.sh]: :+1:

File: fiware-region-sanity-tests/Dockerfile (hunk 4 above)
Comment (jesuspg) [inline fiware-region-sanity-tests/Dockerfile]: no

File: fiware-region-sanity-tests/resources/docker/start.sh (hunk 2 above)
Comment (pratid) [inline fiware-region-sanity-tests/resources/docker/start.sh]: Maybe the `tail` command should be left apart from the start procedure.

File: fiware-region-sanity-tests/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+FROM centos:6.6
+WORKDIR /opt
+RUN yum update -y
+RUN yum install -y gcc git
+RUN yum install -y wget tar
+RUN yum install -y zlib-devel bzip2-devel openssl-devel ncurses-devel sqlite-devel
+RUN wget http://python.org/ftp/python/2.7.6/Python-2.7.6.tgz
+RUN tar -xzf Python-2.7.6.tgz
+RUN cd Python-2.7.6;./configure --prefix=/usr/local --enable-unicode=ucs4 --enable-shared LDFLAGS="-Wl,-rpath /usr/local/lib"
+RUN cd Python-2.7.6;make && make altinstall
```
Comment (flopezag) [inline fiware-region-sanity-tests/Dockerfile]: it is needed to change the directory again...

File: fiware-region-sanity-tests/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+RUN virtualenv fihealth --system-site-packages
+RUN mkdir /opt/fihealth/src
+WORKDIR /opt/fihealth/src
+RUN echo "source /opt/fihealth/bin/activate" >>/etc/bashrc
+RUN pip2.7 install python-config
+# install dbus
+RUN wget http://repositories.lab.fiware.org/repo/files/dbus-python-0.84.0.tar.gz
+RUN tar -xf dbus-python-0.84.0.tar.gz
+RUN source /opt/fihealth/bin/activate; cd dbus-python-0.84.0;./configure --prefix=/opt/fihealth
+RUN source /opt/fihealth/bin/activate; cd dbus-python-0.84.0;make && make install
```
Comment (flopezag) [inline fiware-region-sanity-tests/Dockerfile:29]: You need to point again the virtuanenv and change the directory...

File: fiware-region-sanity-tests/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+RUN yum -y install system-config-firewall
+RUN service messagebus start
+
+# install pygobject
+RUN wget http://repositories.lab.fiware.org/repo/files/pygobject-2.20.0.tar.gz
+RUN tar -xf pygobject-2.20.0.tar.gz
+RUN source /opt/fihealth/bin/activate; cd pygobject-2.20.0;./configure --prefix=/opt/fihealth
+RUN source /opt/fihealth/bin/activate; cd pygobject-2.20.0;make && make install
```
Comment (flopezag) [inline fiware-region-sanity-tests/Dockerfile]: Idem

File: fiware-region-sanity-tests/Dockerfile (hunk 7 above)
Comment (jesuspg) [inline fiware-region-sanity-tests/Dockerfile:29]: It is not necessary

Comment (flopezag) [inline fiware-region-sanity-tests/Dockerfile:29]: NTC

**Comment by flopezag** [comment]
LGTM

**Comment by pratid** [comment]
LGTM

**Pull request merged**